open_rarity/models/token.py

```diff
@@ -25,5 +25,42 @@ class Token:
     token_standard: TokenStandard
     metadata: TokenMetadata
 
+    def __init__(
+        self,
+        token_identifier: TokenIdentifier,
+        token_standard: TokenStandard,
+        metadata: TokenMetadata,
+    ):
+        self.token_identifier = token_identifier
+        self.token_standard = token_standard
+        self.metadata = self._normalize_metadata(metadata)
+
+    def _normalize_metadata(self, metadata: TokenMetadata) -> TokenMetadata:
+        def normalize_and_reset(attributes_dict, acc):
+            # Helper function that takes in an attributes dictionary and accumulator
+            # and normalizes both attribute name in the dictionary as the key
+            # and the repeated field inside the <Type>Attribute class
+            for attribute_name, str_attr in attributes_dict.items():
+                normalized_attr_name = self._normalize_attribute_name(
+                    attribute_name
+                )
+                acc[normalized_attr_name] = str_attr
+                if str_attr.name != normalized_attr_name:
+                    str_attr.name = normalized_attr_name
+            return acc
+
+        return TokenMetadata(
+            string_attributes=normalize_and_reset(
+                metadata.string_attributes, {}
+            ),
+            numeric_attributes=normalize_and_reset(
+                metadata.numeric_attributes, {}
+            ),
+            date_attributes=normalize_and_reset(metadata.date_attributes, {}),
+        )
+
+    def _normalize_attribute_name(self, attribute_name: str) -> str:
```

Comment:
can we move this to `string_utils.py` ? Should be the same method tbh in both cases.

Reply:
yeah i thought about this too but i thought it was weird for a dataclass to use a shared utils method - thoughts?

Reply:
Okay thought about it and wasn't 100% of the folder directory structure (so feel free to give alternative suggestions) but i placed it in a new file under models called `models/utils/attribute_utils.py` since this function is only used for attribute normalization

Reply:
I wanted to keep it under `models` so we can easily build just the models directory and have no need for outside dependencies if we ever wanted to split up resolver and runtime library